src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java

```diff
@@ -93,57 +94,42 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
             {
                 final String message = "Failed to read historical content which is empty.";
                 logger.error( message );
-                return responseHelper.fromResponse( message );
+                return fromResponse( message );
             }
         }
         catch ( final IOException e )
         {
             final String message = "Failed to read historical content file from request body.";
             logger.error( message, e );
-            return responseHelper.fromResponse( message );
+            return fromResponse( message );
         }
 
-        try
-        {
-            Map<String, String> downloadPaths = reader.readPaths( content );
-            controller.downloadArtifacts( downloadPaths, content );
-            Optional<File> archive = controller.generateArchive( content );
-            if ( archive.isEmpty() )
+        Handler<AsyncResult<Message<Boolean>>> resultHandler = messageAsyncResult -> {
+            if ( messageAsyncResult.result().body() )
             {
-                final String message = "Failed to get downloaded contents for archive.";
-                logger.error( message );
-                return responseHelper.fromResponse( message );
+                logger.info( "Archive generate event: {} is completed, build config id: {}", EVENT_GENERATE_ARCHIVE,
+                             content.getBuildConfigId() );
             }
-            controller.renderArchive( archive.get(), content.getBuildConfigId() );
-        }
-        catch ( InterruptedException e )
-        {
-            final String message = "Artifacts downloading is interrupted.";
-            logger.error( message, e );
-            return responseHelper.fromResponse( message );
-        }
-        catch ( final ExecutionException e )
-        {
-            final String message = "Artifacts download execution manager failed.";
-            logger.error( message, e );
-            return responseHelper.fromResponse( message );
-        }
-        catch ( final IOException e )
-        {
-            final String message = "Failed to generate historical archive from content.";
-            logger.error( message, e );
-            return responseHelper.fromResponse( message );
-        }
-        return Response.created( uriInfo.getRequestUri() ).build();
+            else
+            {
+                logger.error( "Archive generate event: {} is failed, build config id: {}", EVENT_GENERATE_ARCHIVE,
+                              content.getBuildConfigId() );
+            }
+        };
+        bus.request( EVENT_GENERATE_ARCHIVE, content, resultHandler );
+        return Uni.createFrom()
+                  .item( accepted().type( MediaType.TEXT_PLAIN )
+                                   .entity( "Archive created request is accepted." )
+                                   .build() );
```

Comment:
I'm thinking how to check if this archive is generated successfully. Maybe we need another rest endpoint to check this. So we should do:
* assign each generation task an id
* A new api to check the archive generation result by the task id.

Reply:
@ligangty Yeah, I'm also thinking about this, currently users could check the log output to confirm the generation result, taking in the handler of AsyncResult. Another point is the sidecar will check the archive availability before per download, if the archive is empty or content compressed is not available, the sidecar will doProxy as simple.

Sure I agree that we could add the REST of checking result status to make users convenient to notice the generation result, I'll file a story for this. Thanks!